### PR TITLE
Add unittests. Make it running on localhost

### PR DESCRIPTION
**Pull request by Dead-fisher**
This is the second pr.
We add full unittest. Also, we fix bugs on local running.

### CODE REVIEW

**Comment by amcadmus** [comment]
> This is the second pr.
> We add full unittest. Also, we fix bugs on local running.

What is "local running"? do you mean examples?

**Comment by amcadmus** [comment]
Please setup CI for an automatic unittest whenever a new commit is merged in the repository.

File: tests/test_gen_rid.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        self.test_rid_out = "tmp_rid/test_out"
+        self.benchmark_json_dir = "benchmark_json"
+        self.benchmark_mol_dir = "benchmark_mol"
+        self.json_file = os.path.join(self.benchmark_json_dir, "rid.json")
+        # self.benchmark_dir = "benchmark_rid"
+    
+    def test_gen_rid(self):
+        
+        rid.gen_rid(out_dir=self.test_rid_out, mol_dir=self.benchmark_mol_dir, rid_json=self.json_file)
+        self.assertTrue(os.path.exists(self.test_rid_out))
```
Comment (amcadmus) [inline tests/test_gen_rid.py:24]: Is there any file or sub-directory in `self.test_rid_out`? If any we may also want to test the files and sub-directories are successfully generated.

File: tests/test_gen_rid.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        pass
```
Comment (amcadmus) [inline tests/test_gen_rid.py:25]: Why do we need a `pass` here?

**Review by amcadmus — COMMENTED** [review]
I would not recommend adding binary files like `graph.pb` for the unittest cases, because we cannot track the changes in the binary files. 

One can convert the `.pb` files to text file `.pbtxt`, and track them in git. When running the rid models, one can convert them back to `.pb` files. The codes for the conversions can be found in https://github.com/deepmodeling/deepmd-kit/blob/69df1d2e897e5f2109e817784c55d2dffcc479ea/source/train/convert_to_13.py#L13-L26

File: tests/test_make_res.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        enhc_file = ["cls.sel.out", "cluster_threshold.dat", "num_of_cluster.dat", "sel.out"]
+        for ii in sel_dir:
+            os.chdir(ii)
+            for rf in res_file:
+                self.assertTrue(os.path.exists(rf))
+            os.chdir("..")
+        os.chdir(self.cwd)
+        os.chdir(self.test_dir_0 + "/iter.000000/00.enhcMD/000")
+        for ef in enhc_file:
+            self.assertTrue(os.path.exists(ef))
```
Comment (TablewareBox) [inline tests/test_make_res.py:59]: Also comment to similar functions: I think the standard of passing the test is, generated files are exactly the same to pre-generated output files

**Comment by Dead-fisher** [comment]
@amcadmus                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              
Hi Dr.Wang, I have fix some codes, please review them again.

1. I have converted .pb to .pbtxt file, and we can track them now.
2. for manual, I add some details to describe how to write `cv.json` file, please review it again.
3. Actually, `gen_rid` will not generate files or sub-dirs, it is only a function to check files and gen base work path.  Futher, I extract it as a individual file because I think some other features like receiving customed force field etc. can be achived here. If it is found to be useless and unnecessary later, I will delete it.
4. `pass` is what I foraget to delete after debug, now i have removed them all.

File: tests/test_make_res.py (hunk 3 above)
Comment (Dead-fisher) [reply tests/test_make_res.py:67]: Yes, I agree with you, but it is not urgent now.......anyway, i will fix them next stage.

**Review by amcadmus — APPROVED** [review]

**Review by TablewareBox — APPROVED** [review]

**Review by dongdawn — APPROVED** [review]

**Pull request merged**